app/assets/src/components/common/DetailsSidebar/SampleDetailsMode/SampleDetailsMode.jsx

```diff
@@ -91,15 +97,26 @@ class SampleDetailsMode extends React.Component {
       return;
     }
 
-    this.setState({
-      metadata: set(key, value, this.state.metadata),
-      metadataChanged: set(key, !shouldSave, this.state.metadataChanged),
-      metadataErrors: set(key, null, this.state.metadataErrors)
-    });
+    this.setState(
+      {
+        metadata: set(key, value, this.state.metadata),
+        metadataChanged: set(key, !shouldSave, this.state.metadataChanged),
+        metadataErrors: set(key, null, this.state.metadataErrors)
+      },
+      () => {
+        if (shouldSave) {
+          this._save(this.props.sampleId, key, value);
```

Comment:
I moved `_save` into `setState` callback to avoid the race condition on `metadata`.